src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -96,7 +102,7 @@ public CompletableFuture<SecurityAnalysisReport> run(String workingVariantId, Se
         Objects.requireNonNull(contingenciesProvider);
         return CompletableFutureTask.runAsync(() -> {
             network.getVariantManager().setWorkingVariant(workingVariantId);
```

Comment:
Is this still necessary  since the runAsync is supposed to do it ?

Reply:
You probably need this to be done before loadContingencies(network).
But it would seem more consistent to me to do that in runAsync since you pass the workingVariantId as param.